### PR TITLE
6.8.0: Document api/agent backend config options and ready API

### LINKED ISSUE #3971: api-serve-wait-time backend config option
### Description

Sensu Go 6.8.0 will include a new `api-serve-wait-time` configuration option (default 0) to allow users to configure a time in seconds to wait on startup before serving API requests.

When the backend is started apid will bind to the api port (8080 by default). With the exception of the /health endpoint, all routes should return an http 503 along with a Retry-After header indicating the api-listen-wait-time (not the time remaining so as to avoid a thundering herd.) After the wait time passes, apid will begin serving traffic as usual.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [x] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-schedule/backend/#backend-configuration-options (list config option)
https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-schedule/backend/#general-configuration (add description table for config option)

### Additional context
Issue: https://github.com/sensu/sensu-go/issues/4826
PR: https://github.com/sensu/sensu-go/pull/4831

### LINKED ISSUE #3972: agent-listen-wait-time backend config option
### Description
Sensu Go 6.8 will include an `agent-listen-wait-time` backend configuration option (default 0) to allow users to configure a time in seconds to wait on startup before accepting agent connections.

When the backend is started agentd will bind to the agent port (8081 by default). When an agent-listen-wait-time is set, the agent websocket route / should return a http 503 until the wait time has expired. After the wait time has expired agentd will begin accepting agent connections as usual. At startup the /health route should begin serving traffic as usual.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [x] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-schedule/backend/#backend-configuration-options (list config option)
https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-schedule/backend/#general-configuration (add description table for config option)

### Additional context
Issue: https://github.com/sensu/sensu-go/issues/4825
PR: https://github.com/sensu/sensu-go/pull/4834

### LINKED ISSUE #3973: /ready API endpoint
### Description
Sensu Go 6.8.0 will include a new endpoint, `/ready`. The new /ready api endpoint will be added to the sensu-go APIs (the sensu-go API as well as the agent listener). /ready will return a HTTP 200 when the API is ready to serve traffic, and will otherwise fail (503).

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [x] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
Add a standalone page for the /ready endpoint under https://docs.sensu.io/sensu-go/latest/api/other/

### Additional context
Issue: https://github.com/sensu/sensu-go/issues/4827
PRs: https://github.com/sensu/sensu-go/pull/4831 and https://github.com/sensu/sensu-go/pull/4834

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Documents the `api-serve-wait-time` and `agent-listen-wait-time` backend configuration options in the backend reference.
- Adds a page in the API category for the /ready API endpoint.
- Adds a reference page for the /ready API endpoint in the Operations > Monitor Sensu sub-category.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/3971
Closes https://github.com/sensu/sensu-docs/issues/3972
Closes https://github.com/sensu/sensu-docs/issues/3973

Signed-off-by: Hillary Fraley <hfraley@sumologic.com>

### CODE REVIEW

File: content/sensu-go/6.8/api/other/ready.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+{{< code shell >}}
+curl -X GET \
+http://127.0.0.1:8081/ready
+{{< /code >}}
+
+The request results in a successful `HTTP/1.1 200 OK` response and a text response body:
+
+{{< code text >}}
+ready
```
Comment (hillaryfraley) [inline content/sensu-go/6.8/api/other/ready.md:75]: Need to confirm after https://github.com/sensu/sensu-enterprise-go/pull/2418 is merged

File: content/sensu-go/6.8/api/other/ready.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+{{< /code >}}
+
+If the backend configuration includes an [`agent-listen-wait-time`][2] duration, the request will result in an `HTTP/1.1 503 Service Unavailable` response.
+Until the `agent-listen-wait-time` duration expires, the text response body will state that agentd is unavailable:
+
+{{< code text >}}
+agentd temporarily unavailable during startup
```
Comment (hillaryfraley) [inline content/sensu-go/6.8/api/other/ready.md:82]: Need to confirm after https://github.com/sensu/sensu-enterprise-go/pull/2418 is merged

File: content/sensu-go/6.8/observability-pipeline/observe-schedule/backend.md [hunk 3]
```diff
@@ -697,6 +712,19 @@ sensu-backend start --agent-host 127.0.0.1{{< /code >}}
...
+| agent-listen-wait-time |      |
+-------------|------
+description  | Time to wait after starting the backend before accepting agent connections. In seconds.
+type         | Integer
+default      | `0`
+environment variable | `SENSU_BACKEND_AGENT_LISTEN_WAIT_TIME`
+command line example   | {{< code shell >}}
+sensu-backend start --agent-listen-wait-time 1{{< /code >}}
+backend.yml config file example | {{< code shell >}}
+agent-listen-wait-time: 1{{< /code >}}
```
Comment (hillaryfraley) [inline content/sensu-go/6.8/observability-pipeline/observe-schedule/backend.md]:
```suggestion
type         | String
default      | `0s`
environment variable | `SENSU_BACKEND_AGENT_LISTEN_WAIT_TIME`
command line example   | {{< code shell >}}
sensu-backend start --agent-listen-wait-time 10s{{< /code >}}
backend.yml config file example | {{< code shell >}}
agent-listen-wait-time: 10s{{< /code >}}
```

File: content/sensu-go/6.8/operations/monitor-sensu/ready.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+{{< code text >}}
+API unavailable during startup.
+See api-serve-wait-time settings.
+{{< /code >}}
+
+Use `agent-listen-wait-time` to configure a delay after startup before the agent listener will begin accepting agent connections.
+The text response body will state that agentd is unavailable until the specified duration expires:
+
+{{< code text >}}
+agentd temporarily unavailable during startup
```
Comment (hillaryfraley) [inline content/sensu-go/6.8/operations/monitor-sensu/ready.md:57]: Need to confirm after https://github.com/sensu/sensu-enterprise-go/pull/2418 is merged

Comment (c-kruse) [reply content/sensu-go/6.8/operations/monitor-sensu/ready.md:57]: @hillaryfraley  merged!

File: content/sensu-go/6.8/api/other/ready.md (hunk 1 above)
Comment (hillaryfraley) [reply content/sensu-go/6.8/api/other/ready.md:75]: Confirmed correct

File: content/sensu-go/6.8/api/other/ready.md (hunk 2 above)
Comment (hillaryfraley) [reply content/sensu-go/6.8/api/other/ready.md:82]: Confirmed

File: content/sensu-go/6.8/operations/monitor-sensu/ready.md (hunk 4 above)
Comment (hillaryfraley) [reply content/sensu-go/6.8/operations/monitor-sensu/ready.md:57]: confirmed

File: content/sensu-go/6.8/api/other/ready.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+output           | 200 (OK):<br><br>{{< code text >}}
+ready
+{{< /code >}}<br>503 (Service Unavailable):<br><br>{{< code text >}}
+API unavailable during startup.
+See api-serve-wait-time settings.
+{{< /code >}}
+
+## Get agent connection readiness data for your Sensu instance
+
+The `/ready` agent transport API endpoint provides HTTP GET access via the backend WebSocket to information about whether your Sensu agent transport is ready to accept agent connections.
```
Comment (c-kruse) [inline content/sensu-go/6.8/api/other/ready.md]: I'd suggest rewording this just a bit or just dropping the "via the backend WebSocket" part.
It isn't a websocket request - it just happens to be served on the same http service that we use to accept agent websocket connections. It looks like we call this service the "agent listener" in most other places.

Maybe something like
> The `/ready` agent listener API endpoint provides HTTP GET access to information about whether your Sensu agent transport is ready to accept agent WebSocket connections.

File: content/sensu-go/6.8/api/other/ready.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+### Example
+
+The following example demonstrates a GET request via the backend WebSocket to the `/ready` agent transport API endpoint using the default WebSocket port 8081:
```
Comment (c-kruse) [inline content/sensu-go/6.8/api/other/ready.md]: Same deal here - "via the backend WebSocket" is a little misleading.

> ... a GET request to the backend agent listener `/ready` API endpoint using the default agent listener port 8081.

File: content/sensu-go/6.8/api/other/ready.md (hunk 5 above)
Comment (hillaryfraley) [reply content/sensu-go/6.8/api/other/ready.md]: Fixed it, thank you!

File: content/sensu-go/6.8/api/other/ready.md (hunk 6 above)
Comment (hillaryfraley) [reply content/sensu-go/6.8/api/other/ready.md]: same fix here!

**Review by c-kruse — APPROVED** [review]
Thank you @hillaryfraley!
